src/Controllers/Common/ContentController.cs

```diff
@@ -2293,4 +2194,106 @@ public IActionResult GetRevealIndex()
             return adjectives[rand.Next(adjectives.Length)] + name;
         return name;
     }
+
+    private CommonInventoryResponse PurchaseItemsImpl(Viking viking, Dictionary<int, int> itemsToPurchase, bool mysteryBox) {
```

Comment:
maybe `addAsMysteryBox` instead of just `mysteryBox`

Reply:
Fixed